test/lsptoolshost/integrationTests/index.ts

```diff
@@ -3,11 +3,73 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import * as vscode from 'vscode';
+import * as fsExtra from 'fs-extra';
+import path from 'path';
+import { AggregatedResult } from '@jest/test-result';
 import { runIntegrationTests } from '../../runIntegrationTests';
 import { jestIntegrationTestProjectName } from './jest.config';
+import { activateCSharpExtension } from './integrationHelpers';
 
 export async function run() {
     process.env.RUNNING_INTEGRATION_TESTS = 'true';
 
-    await runIntegrationTests(jestIntegrationTestProjectName);
+    await activateCSharpExtension();
+    await moveLogs('activated');
+
+    let anyFailures = false;
+    let results: AggregatedResult | undefined;
+    if (process.env.TEST_FILE_FILTER) {
+        results = await runIntegrationTests(jestIntegrationTestProjectName);
+        await moveLogs(path.basename(process.env.TEST_FILE_FILTER, '.integration.test.ts'));
+        anyFailures = anyFailures || !results.success;
+    } else {
+        const workingDirectory = process.cwd();
+
+        const testFiles = (await fsExtra.readdir(__dirname))
+            .filter((file) => file.endsWith('.integration.test.js'))
+            .map((file) => path.join(__dirname, file));
+
+        for (let file of testFiles) {
+            // We have to fix up the file path because the test file was discovered in the /out/ directory.
```

Comment:
do sourcemaps still work if we do this?  e.g. f5 launching integration tests is fine?

Reply:
our launch targets set the `TEST_FILTER_FILE` so it shouldn't run this bit of code.